### PR TITLE
TDD: Test driven development

### PR DESCRIPTION
**Pull request by mateo951**
- Added linters
- Added RSpec to actions
- Added test for creating Solver new instance
- Added code Solver class
- Added test for factorial method
- Added factorial method for solver class
- Added tests for Solver reverse method
- Added code to reverse method
- Added test with tdd for fizzbuzz method
- Added solver functions
- Added readme file
- Updated readme file

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  context 'When testing the fizzbuzz method' do
+    it 'Should return a string' do
+      new_solver = Solver.new
+      expect(new_solver.fizzbuzz(15)).to eq 'fizzbuzz'
+      expect(new_solver.fizzbuzz(9)).to eq 'fizz'
+      expect(new_solver.fizzbuzz(10)).to eq 'buzz'
+      expect(new_solver.fizzbuzz(4)).to eq 'Its not divisble by 3 nor 5'
+    end
+  end
+end
```
Comment (cyonii) [inline spec/solver_spec.rb]: - [x] Make sure to test each method in its own `describe` block for readability and clarity, you tested for the `Solver` class and the `#factorial` method in the same block. For example, the scaffold for your tests should look like 👇🏼 
```ruby
require_relative '../solver'

describe Solver do
    # you can test for the Solver class here
    # you can use contexts if necessary

    describe "#factorial"
        # your assertions here
    end

    describe "#reverse"
        # your assertions here
    end

    describe "#fizzbuzz"
        # your assertions here
    end
end
```
- [x] Remember to test for both positive and negative scenarios, e.g for the `#factorial` method, a negative value should raise an exception.

- [x] [OPTIONAL] For the `#fizzbuzz` method, you can divide the tests into separate `context` blocks for readability and clarity, e.g, you can add `context`s when the number is both divisible by `5` and `3`, when it's only divisible by `5`, when it's only divisible by `3`, and when it's not divisible by neither.
```ruby
describe "#fizzbuzz"
    context "when the number is both divisible by 5 and 3"
        # assertions here
    end

    context "when the number is only divisible by 5"
        # assertions here
    end

    context "when the number is only divisible by 3"
        # assertions here
    end

    context "when the number is neither divisible by 5 nor 3"
        # assertions here
    end
end

**Review by cyonii — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! 👏🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!
Your codebase is quite neat 👏🏼 🍾 

## Required Changes ♻️
_Check the comments under the review to see the changes you need to make._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @cyonii  🙋‍♀️, 

# STATUS APPROVED ✔

Thanks for making the changes 👍  
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (Felipeg005) [reply spec/solver_spec.rb]: Thank you cyonii, now we fix the code for a better readability and clarity 😄

**Pull request merged**